src/components/Table.tsx

```diff
@@ -0,0 +1,177 @@
+import get = require('lodash/get');
+import isPlainObject = require('lodash/isPlainObject');
+import map = require('lodash/map');
+import reverse = require('lodash/reverse');
+import sortBy = require('lodash/sortBy');
+import * as React from 'react';
+import FaSort = require('react-icons/lib/fa/sort');
+import styled from 'styled-components';
+
+import Button from './Button';
+import { Flex } from './Grid';
+import theme from '../theme';
+
+type TableDataItem =
+	| void
+	| string
+	| JSX.Element
+	| {
+			value: any;
+			display: string | JSX.Element;
+		};
+
+export interface TableDataRow {
+	[key: string]: TableDataItem;
+}
+
+type TableSortFunction = <T>(a: T, b: T) => number;
+
+interface TableHeaderItem {
+	field: string;
+	icon?: string;
+	label?: string | JSX.Element;
+	sortable?: boolean | TableSortFunction;
+}
+
+interface TableProps {
+	header: TableHeaderItem[];
+	data: TableDataRow[];
+}
+
+const BaseTable = styled.table`
+	width: 100%;
+	border-spacing: 0;
+
+	> thead {
+		background-color: #f2f2f2;
+
+		> tr > th {
+			padding-left: 40px;
+			padding-top: 10px;
+			padding-bottom: 10px;
+			font-size: 16px;
+		}
+	}
+
+	> tbody {
+		> tr {
+			> td {
+				font-size: 16px;
+				padding-top: 14px;
+				padding-bottom: 14px;
+				padding-left: 40px;
+			}
+
+			&:nth-of-type(even) {
+				background-color: #f8f8f8;
+			}
+		}
+	}
+`;
+
+const HeaderButton = styled(Button)`
+	display: block;
+`
+
+const renderField = (row: TableDataRow, field: string) => {
+	const value = get(row, field);
+
+	if (!value) {
+		return '';
+	}
+
+	return value.hasOwnProperty('display') ? (value as any).display : value;
```

Comment:
I think that this should be `isPlainObject(value) && value.hasOwnProperty('display')`.

Reply:
So, if I get this right, each item value can be an object with a prerendered element as it's value?
In that case, the consumer is responsible to update the data-elements based on state changes?

IMO a render/display function on the column definition is more sensible and easier to reason, but I guess that you already have a use case that this makes huge difference. If that's the case then I'm fine with this.

(Had separation of data & presentation in mind, but I'm not sure whether this fits in a React world.)

Reply:
That's an interesting approach and one I hadn't thought of. The separation of data & presentation does get a bit blurry in the react world for sure.
The reason why I took this approach is because it makes it trivially easy to build interactive UI based on a data item. See here for a slightly messy example --> https://github.com/resin-io/resin-ui/blob/87caac6e2bd808d5ee5652bb3889bef60794f9c3/src/scripts/modules/environment-variables/config-variables.tsx#L234

The idea of having all the presentational information in the header (column) prop and then the data just being raw data is quite attractive.
I'm going to let this brew for an hour and come back to it :)

Reply:
That example is straightforward :+1: